esinstall/src/entrypoints.ts

```diff
@@ -227,34 +205,112 @@ export function resolveEntrypoint(
   );
 }
 
-/**
- * Given an export map and all of the crazy variations, condense down to a key/value map of string keys to string values.
- */
-export function normalizeExportMap(exportField?: ExportField): Record<string, string> | undefined {
-  if (!exportField) {
-    return;
-  }
-  const cleanExportMap: Record<string, string> = {};
+const picoMatchGlobalOptions = Object.freeze({
+  capture: true,
+  noglobstar: true,
+});
+
+function* forEachExportEntry(
+  exportField: ExportField,
+): Generator<[string, unknown], any, undefined> {
+  const simpleExportMap = findExportMapEntry(exportField);
 
-  const simpleExportMap = resolveExportMapEntry(exportField); // handle case where export map is a string, or if there‘s only one file in the entire export map
+  // Handle case where export map is a string, or if there‘s only one file in the entire export map
   if (simpleExportMap) {
-    return {'.': simpleExportMap} as Record<string, string>;
+    yield ['.', simpleExportMap];
+
+    return;
   }
 
   for (const [key, val] of Object.entries(exportField)) {
     // skip invalid entries
     if (!key.startsWith('.')) {
       continue;
     }
-    // skip wildcards for now (https://nodejs.org/api/packages.html#packages_subpath_folder_mappings)
-    if (key.endsWith('/') || key.includes('*')) {
+
+    yield [key, val];
+  }
+}
+
+function* forEachWildcardEntry(
+  key: string,
+  value: string,
+  cwd: string,
+): Generator<[string, string], any, undefined> {
+  try {
+    let expr = pm.makeRe(value, picoMatchGlobalOptions);
+    let reldirnm = path.dirname(value);
+    let dirpth = path.join(cwd, reldirnm);
+    let files = fs.readdirSync(dirpth);
+
+    for (let file of files) {
+      let relfile = path.join(reldirnm, file);
+      let match = expr.exec(relfile);
+
+      if (match && match[1]) {
+        let [pth, repl] = match;
+        let k = key.replace('*', repl);
+        yield [k, './' + pth];
+      }
+    }
+  } catch {
+    // Should we just pretend this didn't happen?
```

Comment:
This happens probably because `fs.readdirSync` threw because the directory didn't exist. Can change this to `existsSync` instead if that's preferred.